app/controllers/theme_controller.py

```diff
@@ -0,0 +1,158 @@
+                    if stylesheet_path.exists():
+                        supported_themes.add(subfolder.name)
+                        logger.info(
+                            f"Found theme with stylesheet in {folder}: {subfolder.name}"
+                        )
+        return supported_themes
+
+    def load_theme(self, theme_name: str) -> Optional[str]:
+        """Load the specified theme."""
+        if theme_name in self.themes:
+            logger.info(f"Loading theme: {theme_name}")
+            stylesheet_path = self.get_theme_stylesheet_path(theme_name)
+            if stylesheet_path:
+                try:
+                    with open(stylesheet_path, "r") as f:
+                        stylesheet = f.read()
+                    return stylesheet
+                except Exception as e:
+                    logger.error(f"Error loading theme: {e}")
+                    return None
+            else:
+                logger.error(f"Stylesheet not found for theme: {theme_name}")
+                return None
+        else:
+            logger.error(f"Attempted to load unsupported theme: {theme_name}")
+            # Revert to the default theme "RimPy"
+            self.theme_name = self.default_theme
+            return self.load_theme(
+                self.default_theme
+            )  # Attempt to load the default theme
+
+    def get_theme_stylesheet_path(self, theme_name: str) -> Optional[Path]:
+        """Returns the path to the stylesheet for the specified theme."""
+        logger.info(f"Searching for stylesheet for theme: {theme_name}")
+
+        # Initialize a variable to hold the path
+        stylesheet_path = None
+
+        for folder in self.theme_stylesheets_folder:
+            # Construct the stylesheet path
+            potential_path = folder / theme_name / "style.qss"
+            logger.debug(f"Checking for stylesheet at: {potential_path}")
+
+            if potential_path.exists():
+                logger.info(
+                    f"Found stylesheet for theme '{theme_name}' at: {potential_path}"
+                )
+                return potential_path  # Return as soon as we find it
+
+            # Update stylesheet_path to the last checked path
+            stylesheet_path = potential_path
+
+        # If we reach this point, it means we didn't find the stylesheet
+        logger.error(
+            f"Stylesheet path does not exist for theme '{theme_name}': {stylesheet_path}"
+        )
+        return None
+
+    def apply_theme(self) -> None:
+        """Applies the current theme by setting the application stylesheet."""
+        stylesheet = self.load_theme(self.theme_name)
+        if stylesheet:
+            logger.info(
+                f"Applying theme: {self.theme_name} with stylesheet: {stylesheet}"
+            )
+            # Implement theme application logic here
+        else:
+            logger.error(f"Failed to apply theme: {self.theme_name}")
```

Comment:
I'm not really sure what this method is for.

Reply:
Removed 
This was for something i was trying a left over